.eslintrc.js

```diff
@@ -55,5 +55,48 @@ module.exports = {
                 case: 'kebabCase',
             },
         ],
+        '@typescript-eslint/naming-convention': [
+            'error',
+            {
+                selector: 'typeLike', // class, interface, typeAlias, enum, typeParameter
+                format: ['PascalCase', 'UPPER_CASE'],
+            },
+            {
+                selector: 'class',
+                format: ['PascalCase'],
+                // prefix: ['Tui', 'Example', 'Test', 'Mock'], // TODO: enable later
+            },
+            {
+                selector: 'function',
+                format: ['camelCase', 'PascalCase'],
+                // prefix: ['tui', 'Tui', 'mock', 'test'], // TODO: enable later
```

Comment:
i have doubts about prefix.
Can we configure the rule to enable prefix-restriction only for exported function/class/etc (only public API) ?

Reply:
about this, there you can configure for non-exported

but in fact everything is exported in the project, so the maximum that I can do is to exclude the demo and scripts directories and spec.ts files

Reply:
fixed